pkg/controller/job/job_controller.go

```diff
@@ -903,9 +908,13 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 		}
 	}
 
-	needsStatusUpdate := suspendCondChanged || active != job.Status.Active || !pointer.Int32Equal(ready, job.Status.Ready)
+	needsStatusUpdate := suspendCondChanged || active != job.Status.Active || !pointer.Int32Equal(job.Status.Ready, ready)
 	job.Status.Active = active
 	job.Status.Ready = ready
+	if (job.Spec.PodFailurePolicy != nil && feature.DefaultFeatureGate.Enabled(features.JobPodFailurePolicy)) || feature.DefaultFeatureGate.Enabled(features.JobPodReplacementPolicy) {
```

Comment:
Are the lookups to PodFailurePolicy still needed here, given that we have a defaulting now? Also, maybe we can have a helper function: `hasPodReplacementPolicyFailed` to abstract it out.

Reply:
updated.